### PR TITLE
Implement Custom TCAM profile support

### LINKED ISSUE #620: Update TCAM support in CLI Config Gen to support custom profile
**Issue Type**

- [x] Template enhancement
- [ ] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

The current TCAM profile only supports which profile to apply to the system. But some features require to implement a custom TCAM profile.

**Describe the solution you'd like**

```yaml
tcam_profile:
  system: < tcam_profile >
  profiles:
    < profile name >: |
      < multiline EOS TCAM profile >
      < multiline EOS TCAM profile >
      < multiline EOS TCAM profile >
```

It can also work with lookup plugins depending of user's workflow

```yaml
tcam_profile:
  system: < tcam_profile >
  profiles:
    < profile name >: "{{lookup('file', '< path to EOS TCAM configuration >')}}"
```

**Describe alternatives you've considered**

Other option would be to load a snippet from a file to use with `< profile name >` but would be a different process compare to other CLI CONFIG GEN things

### PR DESCRIPTION
**Pull request by titom73**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #620

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Change datamodel to support creation of custom TCAM profiles

```yaml
tcam_profile:
  system: < tcam profile name to activate >
  profiles:
    < tcam_profile 01 >: < tcam profile in EOS syntax >
```

## How to test

Look at molecule test

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tcam-profile.j2 [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{%     endif %}
+
+## Hardware TCAM configuration
+
+```eos
+{% include 'eos/tcam-profile.j2' %}
+```
+{% else %}
+Hardware TCAM profile is not defined
+{% endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tcam-profile.j2]: missing newline

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/tcam-profile.j2 [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+{{ tcam_profile.profiles[profile] }}
+   !
+{%         endfor %}
+{%     else %}
+   !
+{%     endif %}
+{%     if tcam_profile.system is arista.avd.defined %}
+   system profile {{ tcam_profile.system }}
+{%     endif %}
 {% endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/tcam-profile.j2]: missing newline

**Review by ClausHolbechArista — DISMISSED** [review]
Very nice implementation!
This will break l3ls platform template, so we should get that fixed to support this new data model.
Approved but missing a few newlines.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tcam-profile.j2 (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tcam-profile.j2]: @titom73 still missing newline :)

**Comment by ClausHolbechArista** [comment]
We need to change eos_designs as part of this PR

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tcam-profile.j2 (hunk 1 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tcam-profile.j2]: Solved with commit 682ea624

**Comment by carlbuchmann** [comment]
@ClausHolbechArista see commit b673d96 - eos_designs updated!

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Comment by ClausHolbechArista** [comment]
Looking good.

**Pull request merged**